### ISSUE TITLE
Expose account-url as a static parameter not just as an env var

### ISSUE THREAD

**Issue by kennyjwilli** [CircleCI-Public/aws-ecr-orb#39]
### Orb version

6.1.0

### What happened

`account-url` is only exposed as an environment variable.

### Expected behavior

`account-url` is exposed as an environment variable **and** a static string.

**Comment by iynere** [comment]
@kennyjwilli this is an intentional design decision, to promote better security. i don't think we want folks having the option to pass in their account URL in plaintext.

**Comment by kennyjwilli** [comment]
Not sure I agree. The account URL simply contains your AWS account ID and an AWS region. Neither of those things needs to be kept secret. For example, Datadog publishes their AWS account ID publicly for their [AWS integration](https://docs.datadoghq.com/integrations/amazon_web_services/?tab=allpermissions#installation). The option should be provided to orb users to determine if passing the account ID as plaintext is ok or not.

**Comment by iynere** [comment]
@kennyjwilli that's fair—feel free to submit a pr to make that modification, should be pretty simple. thank you!

**Comment by stephenh** [comment]
The docs here also use a static string:

https://circleci.com/docs/2.0/ecs-ecr/

```
workflows:
  build-and-deploy:
    jobs:
      - aws-ecr/build_and_push_image:
          account-url: "${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com"
          repo: "${AWS_RESOURCE_NAME_PREFIX}"
          region: ${AWS_DEFAULT_REGION}
          tag: "${CIRCLE_SHA1}"
      - ...
```

As well as this example from the aws-ecs docs:

https://github.com/CircleCI-Public/aws-ecs-orb/blob/master/src/orb.yml.hbs#L22

@iynere I'm happy to do a PR but what's the right yaml incantation to say "this can be either a string or an env-var, and if its an env-var, then resolve it, otherwise leave it alone", i.e. to keep backwards compatibility?

**Comment by KyleTryon** [comment]
The String environment interpolation was a convention we used to use around the time of the launch of orbs as we did not yet have the much safer `env_var_name` parameter type.
https://circleci.com/docs/2.0/reusing-config/#environment-variable-name

This Orb is due for an overhaul to bring us up to date with the latest best practices and design behaviors. https://circleci.com/docs/2.0/orbs-best-practices/#orb-best-practices-guidelines

@stephenh The best practice now is to _not_ allow strings in any case where sensitive information may be exposed. It is a design choice to attempt to restrict users from entering API keys, passwords, etc. 

For _all_ secret information, we ask the user to implement an environment variable and provide the name of said environment variable via the `env_var_name` parameter.

Utilizing this in code looks like:

```
    - run:
        command: |
          s3cmd --access_key ${FOO_BAR} \
                --secret_key ${BIN_BAZ} \
                ls s3://some/where
```

This also opens the door for users to use secret management systems, beyond helping to protect users from exposing private information.

I think it would be a good idea for our team to revisit this orb and re-author it using the new Orb Starter Kit shortly.

https://github.com/CircleCI-Public/orb-starter-kit/pull/37

**Comment by ghostsquad** [comment]
Yes please. A naming convention would also help. `account-url` sounds like I should provide an account url, not an environment variable name that points to the account url.

example: account-url-env
    and: account-url

and you can provide either or, if you provide both, maybe the `account-url` takes precedence, or an error is thrown.

**Comment by ghostsquad** [comment]
urls aren't secret. passwords are secret. So this makes perfect sense for the accesskey and secretkey. But not the other values.

**Comment by ExternalReality** [comment]
While I agree with that url should not be secret, it is also the case that the URL can be derived directly from the region and the account id. The orb, already having requesting the region as a parameter, should simply request the account_id and construct the url internally form that information.

```
 - aws-ecr/build_and_push_image:
          account-account-id: account_id 
          repo: repo_name
          region: region
```

The above is enough information to construct the url.

https://docs.aws.amazon.com/AmazonECR/latest/userguide/Registries.html

**Comment by brivu** [comment]
Hey @ExternalReality, 

I'm Brian from CircleCI's CPE Team.  I just wanted to let you know thatI've taken care of this issue and it'll be merged with the next PR.

Thanks for your contribution! 
Brian
